src/v/cloud_storage/topic_manifest.h

```diff
@@ -74,5 +89,6 @@ class topic_manifest final : public base_manifest {
 
     std::optional<cluster::topic_configuration> _topic_config;
     model::initial_revision_id _rev;
+    version_t _manifest_version{first_version};
```

Comment:
I don't think there's anything semantically different about the `topic_manifest` between `first_version` and `serde_version`. Why not just store a `manifest_format` and expose that?

Reply:
not sure i understand the question,

i think there is a semantic difference: first_version encoded only 7 fields from topic_properties, the user code is aware of this and has to fill a bunch of properties depending on how it will use topic_configuration.

serde_version makes it possible to ignore the fact that a topic_property is missing some newer properties, because these will be filled with the default values and the gap of missing properties should be small.
for example in topic_recovery_service right now it has to set the local retention properties, but whith the new version it can trust the topic_manifest

so the user code needs to know what is the source of the topic_manifest, if it has to expect that all the properties after retention_duration are missing or if it can trust that only most recent one might be missing. special rules (like setting recovery=true) still apply though

Reply:
Ah I see. I kind of buy that, but then I don't see `get_manifest_version()` being used at all, so it isn't clear where that's happening.

Reply:
It was used in the previous designs.
I haven't attached yet the commits using it, I'm thinking now of having them in a separate PR just to not mix things up here